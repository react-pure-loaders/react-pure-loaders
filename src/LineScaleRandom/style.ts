import { keyframes } from '@emotion/react';
import { globalAnimation, lines, ballGrid } from '../mixins';

const random = 0.5;

const ballBeat = keyframes({
  '0%': {
    transform: 'scale(1.0)',
  },
  '100%': {
    transform: 'scale(1.0)',
  },
  '50%': {
    transform: `scale(${random})`,
  },
});

export default (color?: string) => ({
  ...ballGrid(),
  ' > div': {
    ...lines(color),
    ...globalAnimation(),
    animationDelay: '0',
    animationIterationCount: 'infinite',
    animationName: ballBeat,
    display: 'inline-block',
  },
});
