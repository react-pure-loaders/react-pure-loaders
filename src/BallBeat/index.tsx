/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
