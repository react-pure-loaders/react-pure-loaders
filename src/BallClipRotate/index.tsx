/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
