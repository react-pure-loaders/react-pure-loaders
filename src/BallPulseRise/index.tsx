/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
