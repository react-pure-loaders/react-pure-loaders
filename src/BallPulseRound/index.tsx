/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
