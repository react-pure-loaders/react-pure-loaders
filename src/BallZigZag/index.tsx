/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
