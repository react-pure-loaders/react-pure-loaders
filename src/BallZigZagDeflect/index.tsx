/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
