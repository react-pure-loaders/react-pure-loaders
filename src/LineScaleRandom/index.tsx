/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
