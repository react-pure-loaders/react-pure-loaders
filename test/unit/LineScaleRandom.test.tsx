import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import LineScaleRandom from '../../src/LineScaleRandom';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();
const mathCopy = Object.create(global.Math);

describe('<LineScaleRandom>', () => {
    beforeEach(() => {
        global.Math.random = () => 0.5;
    });

    afterEach(() => {
        global.Math = mathCopy;
        cleanup();
    });

    test('LineScaleRandom should match snapshot', () => {
        const { container } = render(<LineScaleRandom loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('LineScaleRandom should have default color', () => {
        const { container } = render(<LineScaleRandom loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test('LineScaleRandom should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<LineScaleRandom color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('LineScaleRandom should have no children', () => {
        const { container } = render(<LineScaleRandom loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('LineScaleRandom should have three children', () => {
        const { container } = render(<LineScaleRandom loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(5);
    });
});
