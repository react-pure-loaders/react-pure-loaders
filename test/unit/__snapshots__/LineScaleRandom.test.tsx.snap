// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineScaleRandom> LineScaleRandom should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: scale(1.0);
    -moz-transform: scale(1.0);
    -ms-transform: scale(1.0);
    transform: scale(1.0);
  }

  100% {
    -webkit-transform: scale(1.0);
    -moz-transform: scale(1.0);
    -ms-transform: scale(1.0);
    transform: scale(1.0);
  }

  50% {
    -webkit-transform: scale(0.5);
    -moz-transform: scale(0.5);
    -ms-transform: scale(0.5);
    transform: scale(0.5);
  }
}

.emotion-0>div:nth-of-type(1) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 1.1s;
  animation-duration: 1.1s;
}

.emotion-0>div:nth-of-type(2) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 1.1s;
  animation-duration: 1.1s;
}

.emotion-0>div:nth-of-type(3) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 1.1s;
  animation-duration: 1.1s;
}

.emotion-0>div:nth-of-type(4) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 1.1s;
  animation-duration: 1.1s;
}

.emotion-0>div:nth-of-type(5) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 1.1s;
  animation-duration: 1.1s;
}

.emotion-0>div:nth-of-type(6) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 1.1s;
  animation-duration: 1.1s;
}

.emotion-0>div:nth-of-type(7) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 1.1s;
  animation-duration: 1.1s;
}

.emotion-0>div:nth-of-type(8) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 1.1s;
  animation-duration: 1.1s;
}

.emotion-0>div:nth-of-type(9) {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
  -webkit-animation-duration: 1.1s;
  animation-duration: 1.1s;
}

.emotion-0>div {
  border-radius: 2px;
  height: 35px;
  margin: 2px;
  width: 4px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-delay: 0;
  animation-delay: 0;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  display: inline-block;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
